homeassistant/components/verisure/coordinator.py

```diff
@@ -56,60 +62,115 @@ async def async_login(self) -> bool:
 
     async def async_logout(self, _event: Event) -> None:
         """Logout from Verisure."""
-        try:
-            await self.hass.async_add_executor_job(self.verisure.logout)
-        except VerisureError as ex:
-            LOGGER.error("Could not log out from verisure, %s", ex)
 
     async def _async_update_data(self) -> dict:
         """Fetch data from Verisure."""
+        try:
+            await self.hass.async_add_executor_job(self.verisure.update_cookie)
+        except (VerisureLoginError, VerisureResponseError) as ex:
+            LOGGER.error("Credentials expired for Verisure, %s", ex)
+            raise ConfigEntryAuthFailed("Credentials expired for Verisure") from ex
         try:
             overview = await self.hass.async_add_executor_job(
-                self.verisure.get_overview
+                self.verisure.request,
+                self.verisure.arm_state(),
+                self.verisure.broadband(),
+                self.verisure.cameras(),
+                self.verisure.climate(),
+                self.verisure.door_window(),
+                self.verisure.smart_lock(),
+                self.verisure.smartplugs(),
             )
-        except VerisureResponseError as ex:
-            LOGGER.error("Could not read overview, %s", ex)
-            if ex.status_code == HTTPStatus.SERVICE_UNAVAILABLE:
-                LOGGER.info("Trying to log in again")
-                await self.async_login()
-                return {}
-            raise
+        except VerisureResponseError as err:
+            LOGGER.debug("Cookie expired or service unavailable, %s", err)
+            overview = self._overview
+            try:
+                await self.hass.async_add_executor_job(self.verisure.update_cookie)
+            except VerisureResponseError as ex:
+                raise ConfigEntryAuthFailed("Credentials for Verisure expired.") from ex
+            except Exception as ex:
+                LOGGER.error("Could not read overview, %s", ex)
+                raise UpdateFailed("Could not read overview") from ex
+        except VerisureError as err:
+            LOGGER.error("Could not read overview, %s", err)
+            raise UpdateFailed("Could not read overview") from err
+
+        def unpack(overview: list, value: str) -> dict | list:
+            unpacked = [
+                item["data"]["installation"][value]
+                for item in overview
+                if value in item.get("data", {}).get("installation", {})
+            ]
+            return unpacked[0]
```

Comment:
If we only return the first item in the list, why do we unpack them all in that case? Sounds rather inefficient.Find the first item in the overview in that case and unpack that.

Reply:
@frenck Would this solution by @persandstrom satisfy your request?